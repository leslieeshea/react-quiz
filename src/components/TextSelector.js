import React from 'react';
import PropTypes from 'prop-types';

function TextSelector({ text, handleChange }) {
  return (
    <input
      name="text"
      type="text"
      value={text}
      onChange={handleChange}
    />
  );
}

TextSelector.propTypes = {
  text: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default TextSelector;
