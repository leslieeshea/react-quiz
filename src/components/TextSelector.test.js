import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('TextSelector component', () => {
  it('renders a TextSelector', () => {
    const wrapper = shallow(<TextSelector />);
    expect(wrapper).toMatchSnapshot();
  });
});
