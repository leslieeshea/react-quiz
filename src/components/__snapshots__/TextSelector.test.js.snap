// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelector component renders a TextSelector 1`] = `
<input
  name="text"
  type="text"
/>
`;
